### ISSUE TITLE
Framework Update 1.10 failed

### ISSUE THREAD

**Issue by bieberjz** [Icinga/icinga-powershell-framework#574]
An error occurs when updating the Powershell framework. The update is performed via the Icinga for Windows Manager.

## Expected Behavior
The framework update should be updated with the Icinga for Windows Manager. 

## Current Behavior
**Framework Update returns the following error code:** 

> [Notice]: Unblocking Icinga PowerShell Files
> At C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\cache\framework_cache.psm1:20299 char:43
> +         if ($InputObject -Match "[Ã¤Ã¶Ã¼Ã„Ã–ÃœÃŸ]") {
> +                                           ~~~~~~~~~~~
> Unexpected token 'Ã–ÃœÃŸ]") {
>             return $InputObject;
>         }
>         $InputInBytes = [System.Text.Encoding]::Default.GetBytes($InputObject);
>         return ([string]([System.Text.Encoding]::UTF8.GetString($InputInBytes)));
>     } elseif ($InputObject -Is [array]) {
>         [array]$ArrayObject = @();
>         foreach ($entry in $InputObject) {
>             if ($entry -Is [array]) {
>                 $ArrayObject += , (ConvertTo-IcingaUTF8Value -InputObject $entry);
>             } else {
>                 $ArrayObject += ConvertTo-IcingaUTF8Value -InputObject $entry;
>             }
>         }
>         return $ArrayObject;
>     }
>     # If we are not a string or a array, just return the object
>     return $InputObject;
> }
> <#
> .SYNOPSIS
>    Helper function to convert values to integer if possible
> .DESCRIPTION
>    Converts an input value to integer if possible in any way. Otherwise it will return the object unmodified
>    More Information on https://github.com/Icinga/icinga-powershell-framework
> .FUNCTIONALITY
>    Converts an input value to integer if possible in any way. Otherwise it will return the object unmodified
> .PARAMETER Value
>    Any value/object is analysed and if possible converted to an integer
> .INPUTS
>    System.Object
> .OUTPUTS
>    System.Integer

............

[Error]: Failed to enter menu "Show-IcingaForWindowsMenuUpdateComponents". Error "The 'Use-Icinga' command was found in
the module 'icinga-powershell-framework', but the module could not be loaded. For more information, run 'Import-Module i
cinga-powershell-framework'.

**Show-IcingaForWindowMenuUpdateComponents:** 

> The property 'LastValues' cannot be found on this object. Verify that the property exists and can be set.
> At C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\cache\framework_cache.psm1:11940 char:13
> +             $global:Icinga.InstallWizard.LastValues = @();
> +             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
>     + FullyQualifiedErrorId : PropertyNotFound
> 
> The property 'Menu' cannot be found on this object. Verify that the property exists and can be set.
> At C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\cache\framework_cache.psm1:11946 char:13
> +             $global:Icinga.InstallWizard.Menu = $LastCommand;
> +             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
>     + FullyQualifiedErrorId : PropertyNotFound
> 
> You cannot call a method on a null-valued expression.
> At C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\cache\framework_cache.psm1:12439 char:9
> +     if ($global:Icinga.InstallWizard.Config.ContainsKey($InstallerSte ...
> +         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
>     + FullyQualifiedErrorId : InvokeMethodOnNull
> 
> You cannot call a method on a null-valued expression.
> At C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\cache\framework_cache.psm1:12465 char:9
> +     if ($global:Icinga.InstallWizard.Config.ContainsKey($Step) -eq $F ...
> +         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
>     + FullyQualifiedErrorId : InvokeMethodOnNull
> 
> Cannot index into a null array.
> At C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\cache\framework_cache.psm1:12469 char:9
> +     if ($null -eq $global:Icinga.InstallWizard.Config[$Step].Values)  ...
> +         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
>     + FullyQualifiedErrorId : NullArray
> 
> Cannot index into a null array.
> At C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\cache\framework_cache.psm1:12473 char:12
> +     return [array]($global:Icinga.InstallWizard.Config[$Step].Values) ...
> +            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
>     + FullyQualifiedErrorId : NullArray
> 
> The property 'ParentConfig' cannot be found on this object. Verify that the property exists and can be set.
> At C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\cache\framework_cache.psm1:11958 char:5
> +     $global:Icinga.InstallWizard.ParentConfig = $ParentConfig;
> +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
>     + FullyQualifiedErrorId : PropertyNotFound
> 
> The property 'LastInput' cannot be found on this object. Verify that the property exists and can be set.
> At C:\Program Files\WindowsPowerShell\Modules\icinga-powershell-framework\cache\framework_cache.psm1:11959 char:5
> +     $global:Icinga.InstallWizard.LastInput    = $null;
> +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
>     + FullyQualifiedErrorId : PropertyNotFound

## Possible Solution
Convert file ConvertTo-IcingaUTF8Value.psm1 with UTF-8 with BOM and then save it again.

## Steps to Reproduce (for bugs)

1. Open Icinga for Windows Management
2. 1 - Component Manager
3. 2- Update
4. Framework update

## Your Environment

PSVersion                      5.1.14393.5127
Version    BuildNumber Caption
-------    ----------- -------
10.0.14393 14393       Microsoft Windows Server 2016 Standard

**Comment by LordHepipud** [comment]
Thank you for the issue. Did you have a look on the Upgrading docs regarding the upgrade to v1.10.0?

https://icinga.com/docs/icinga-for-windows/latest/doc/100-General/01-Upgrading/

The mentioned exception is expected. For machines being updated already, we added a Knowledge Base entry [IWKB000014]/(https://icinga.com/docs/icinga-for-windows/latest/doc/knowledgebase/IWKB000014/) to resolve the issue.

I have updated the release notes to point directly to the Upgrading docs for more clarification about this.

Afterwards the environment should work again.
